src/java.base/share/classes/java/text/ListFormat.java

```diff
@@ -0,0 +1,600 @@
+     *     <td>""</td>
+     * </tbody>
+     * </table>
+     * Here are the resulting formatted strings with the above pattern array.
+     * <table class="striped">
+     * <caption style="display:none">Formatting examples</caption>
+     * <thead>
+     * <tr><th scope="col">Input String List</th>
+     *     <th scope="col">Formatted String</th></tr>
+     * </thead>
+     * <tbody>
+     * <tr><th scope="row" style="text-align:left">"Foo", "Bar", "Baz", "Qux"</th>
+     *     <td>"Foo, Bar, Baz, and Qux"</td>
+     * <tr><th scope="row" style="text-align:left">"Foo", "Bar", "Baz"</th>
+     *     <td>"Foo, Bar, and Baz"</td>
+     * <tr><th scope="row" style="text-align:left">"Foo", "Bar"</th>
+     *     <td>"Foo and Bar"</td>
+     * <tr><th scope="row" style="text-align:left">"Foo"</th>
+     *     <td>"Foo"</td>
+     * </tbody>
+     * </table>
+     *
+     * @param patterns array of patterns, not null
+     * @throws IllegalArgumentException if the length {@code patterns} array is not 5, or
+     *          any of {@code start}, {@code middle}, {@code end}, {@code two}, or
+     *          {@code three} patterns cannot be parsed.
+     * @throws NullPointerException if {@code patterns} is null.
+     */
+    public static ListFormat getInstance(String[] patterns) {
+        Objects.requireNonNull(patterns);
+        if (patterns.length != PATTERN_ARRAY_LENGTH) {
+            throw new IllegalArgumentException("Pattern array length should be " + PATTERN_ARRAY_LENGTH);
+        }
+        return new ListFormat(Locale.ROOT, Arrays.copyOf(patterns, PATTERN_ARRAY_LENGTH));
+    }
+
+    /**
+     * {@return the string that consists of the input strings, concatenated with the
+     * patterns of this {@code ListFormat}}
+     * @apiNote Formatting the string from an excessively long list may exceed memory
+     *          or string sizes.
+     * @param input The list of input strings to format. There should at least
+     *              one String element in this list, otherwise an {@code IllegalArgumentException}
+     *              is thrown.
+     * @throws IllegalArgumentException if the length of {@code input} is zero.
+     * @throws NullPointerException if {@code input} is null.
+     */
+    public String format(List<String> input) {
+        Objects.requireNonNull(input);
+
+        return format(input, new StringBuffer(),
+                DontCareFieldPosition.INSTANCE).toString();
+    }
+
+    /**
+     * Formats an object and appends the resulting text to a given string
+     * buffer. The object should either be a List or an array of Objects.
+     *
+     * @apiNote Formatting the string from an excessively long list or array
+     *          may exceed memory or string sizes.
+     * @param obj    The object to format. Must be a List or an array
+     *               of Object.
+     * @param toAppendTo    where the text is to be appended
+     * @param pos    Ignored. Not used in ListFormat. May be null
```

Comment:
Curious, why not used? 
I could see a use to identity the string inserted to enable highlighting or other markup around the new string.

Reply:
`FieldPosition` is dedicated to identifying fields in *Format classes which are either `Format.Field` or fields that have names ending with "_FIELD", which ListFormat has neither of them.